out_writeapi.go

```diff
@@ -612,18 +644,22 @@ func FLBPluginExitCtx(ctx unsafe.Pointer) int {
 		log.Printf("Finding configuration for output instance with id: %d failed in FLBPluginExitCtx", id)
 		return output.FLB_ERROR
 	}
+	streamSlice := *config.managedStreamSlice
 
-	checkResponses(ms_ctx, config.appendResults, true, &config.mutex, config.exactlyOnce, id)
+	sliceLen := len(streamSlice)
+	for i := 0; i < sliceLen; i++ {
+		checkResponses(ms_ctx, (streamSlice)[i].appendResults, false, &config.mutex, config.exactlyOnce, id)
 
-	if config.managedStream != nil {
-		if config.exactlyOnce {
-			if _, err := config.managedStream.Finalize(ms_ctx); err != nil {
-				log.Printf("Finalizing managed stream for output instance with id: %d failed in FLBPluginExit: %s", id, err)
+		if streamSlice[i].managedstream != nil {
+			if config.exactlyOnce {
+				if _, err := streamSlice[i].managedstream.Finalize(ms_ctx); err != nil {
+					log.Printf("Finalizing managed stream for output instance with id %d and stream index %d failed in FLBPluginExit: %s", id, i, err)
+				}
+			}
+			if err := streamSlice[i].managedstream.Close(); err != nil {
+				log.Printf("Closing managed stream for output instance with id %d and stream index %d failed in FLBPluginExitCtx: %s", id, i, err)
+				return output.FLB_ERROR
```

Comment:
I'm wondering if we should hold off on returning the error until we are done closing all the streams. So, just set a flag that we encountered this error, but then return the error after the loop is done.

Reply:
Sounds good and done!